### PR TITLE
Postcode work

### PR DESCRIPTION
**Pull request by DenisKent**
To be looked at after PR #98 

added postcode and calendar search functionality.

The cook mamma Affifi lives at the post code N19 4AH and is willing to walk 3km. She is cooking two meals on 21st September 2018. Putting in any postcode within 3km of mamma Affifi on this day will update redux with a mealList.

I have not done a redirect to mealListPage or readthe state in mealList page. Some research is required to understand what the best way of redirecting is based on redux state.

### CODE REVIEW

File: server/database/getQueries/getAvailableMeals.js [hunk 1]
```diff
@@ -30,7 +30,11 @@ const getMealList = () => {
   ON
     meal.cook_user_id = av_review.cook_user_id;
     `;
-  return db.any(queryString);
+  return db.any(queryString, [mealDate]);
 };
+// console.log(new Date(2018, 8, 21));
```
Comment (matthall8) [inline server/database/getQueries/getAvailableMeals.js]: Commented code.

File: server/utils/geoLocation.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const geolib = require('geolib');
+
+const filterLocalMeals = (mealArray, { latitude: cusLat, longitude: cusLong }) => mealArray.filter((elem) => {
+  const { kitchen_lat_long: { x: cookLat, y: cookLong }, delivery_range_m: maxDelivery } = elem;
+  const distance = geolib.getDistance(
+    { latitude: cookLat, longitude: cookLong },
+    { latitude: cusLat, longitude: cusLong },
+  );
+  console.log('customer', cusLat, cusLong);
```
Comment (matthall8) [inline server/utils/geoLocation.js]: Remove console.log!

File: client/src/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+import findMeals from '../../actions/findMeals'
 
+const HomePage = (props) => {
+  const submit = (values) => {
+    const {postcode,deliveryDate} = values
+    axios(`http://api.postcodes.io/postcodes/${postcode}`)
+    .then(({data:{result:location}})=> {
+      props.findMeals({location, deliveryDate})
+    })
+    .catch(console.log)
```
Comment (shiryz) [inline client/src/pages/HomePage/HomePage.js]: You need to display a message to the user that there was an error, could open an issue about it as a stretch goal

File: client/src/pages/HomePage/HomePage.js [hunk 4]
```diff
@@ -21,4 +34,19 @@ const HomePage = () => {
...
+const mapDispatchToProps = {
+  findMeals
+};
+
+export default connect(
+  null,
+  mapDispatchToProps
+)(HomePage);
+
+export default HomePage;
```
Comment (shiryz) [inline client/src/pages/HomePage/HomePage.js]: add new line here

File: client/src/pages/index.js [hunk 5]
```diff
@@ -5,4 +5,4 @@ import HomePage from "./HomePage/HomePage"
 import CheckoutPage from "./CheckoutPage/CheckoutPage"
 
 // eslint-disable-next-line import/prefer-default-export
-export { MealPage, MealsListPage, BasketPage, HomePage, CheckoutPage };
+export { MealPage, MealsListPage, BasketPage, HomePage, CheckoutPage };
```
Comment (shiryz) [inline client/src/pages/index.js]: add new line

File: server/app.js [hunk 6]
```diff
@@ -1,13 +1,15 @@
 const express = require('express');
 const path = require('path');
-
+const bodyParser = require('body-parser');
 const controllers = require('./controllers/index.js');
 
 const app = express();
 app.use(require('body-parser').text());
```
Comment (shiryz) [inline server/app.js]: body-parser required before

File: server/database/getQueries/getAvailableMeals.js [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
 
-const getMealList = () => {
+const getAvailableMeals = (mealDate) => {
   const queryString = `
   SELECT
-    meal.id, meal.meal_title, meal.price, meal.final_booking_at, meal.remaining_portions, meal_thumbnails.meal_image_url, cook_info.cook_firstname, cook_thumbnails.cook_image_url, av_review.av_star_rating, av_review.count_reviews
+    meal.id, meal.meal_title, meal.price, meal.final_booking_at, meal.remaining_portions, date_trunc('day',meal.meal_scheduled_at), meal_thumbnails.meal_image_url, cook_info.cook_firstname, cook_info.kitchen_lat_long, cook_info.delivery_range_m, cook_thumbnails.cook_image_url, av_review.av_star_rating, av_review.count_reviews
   FROM
-    (SELECT * FROM meals) AS meal
+    (SELECT * FROM meals WHERE final_booking_at > NOW() AND date_trunc('day',meal_scheduled_at) = date_trunc('day',$1::timestamp)) AS meal
```
Comment (shiryz) [inline server/database/getQueries/getAvailableMeals.js]: Niice

File: client/src/pages/HomePage/HomePage.js (hunk 4 above)
Comment (shiryz) [reply client/src/pages/HomePage/HomePage.js]: shouldn't export twice

File: client/src/pages/HomePage/HomePage.js [hunk 8]
```diff
@@ -1,10 +1,23 @@
...
+import { connect } from "react-redux";
+import axios from "axios";
 import HomeHeader from './sections/HomeHeader/HomeHeader';
 import MealSearchForm from './sections/MealSearchForm/MealSearchForm'
 import {Footer} from '../../components'
 import {FooterWrapper} from '../../components/Footer/FooterWrapper'
 import {StyledPara} from './HomePage.style'
+import findMeals from '../../actions/findMeals'
 
+const HomePage = (props) => {
```
Comment (shiryz) [inline client/src/pages/HomePage/HomePage.js]: curly braces are not closed in this line and the next and it's not compiling

File: client/src/pages/HomePage/HomePage.js (hunk 3 above)
Comment (matthall8) [reply client/src/pages/HomePage/HomePage.js]: Done here: #73

**Comment by matthall8** [comment]
Done. Please can you review again?

**Pull request merged**